File: benchmarks/fixtures/webapp_csharp/Auth/TokenClaims.cs
namespace Webapp.Auth;

/// <summary>
/// Decoded claims carried by an authentication token.
/// </summary>
public class TokenClaims
{
    public string UserId { get; set; }
    public string Email { get; set; }
    public string Role { get; set; }

    public TokenClaims(string userId, string email, string role)
    {
        UserId = userId;
        Email = email;
        Role = role;
    }
}

File: benchmarks/fixtures/webapp_c/auth/token_claims.c
#include "auth/token_claims.h"

struct TokenClaims token_claims_new(const char *user_id, const char *email, const char *role)
{
    struct TokenClaims claims;
    claims.user_id = user_id;
    claims.email = email;
    claims.role = role;
    claims.issued_at = 0;
    return claims;
}
